Project-Aurora/Project-Aurora/Devices/YeeLight/YeeLightAPI.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            try
+            {
+                yeelightTcpClient.GetStream().Close();
+                yeelightTcpClient.Close();
+
+                yeelightTcpClient = new TcpClient();
+                return true;
+            }
+            catch (Exception exc) { lastError = exc; }
+            return false;
+        }
+
+        private bool SendString(string command)
+        {
+            try
+            {
+                if (yeelightTcpClient.Connected)
+                {
+                    return yeelightTcpClient.Client.Send(Encoding.ASCII.GetBytes(command)) > 0;
+                }
+            }
+            catch (Exception exc) { lastError = exc; }
+            return false;
+        }
+
+        public bool SetBrightness(int brightness) //Range is 0 - 100
+        {
+            try
+            {
+                if (yeelightTcpClient.Connected && brightness <= 100 && brightness >= 0)
+                {
+                    return SendString("{\"id\":" + 1 + ",\"method\":\"set_bright\",\"params\":[ " + brightness.ToString() + ", \"sudden\", 100" + "]}\r\n");
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception exc) { lastError = exc; }
+            return false;
+        }
+
+        public bool SetPower(bool status)
+        {
+            try
+            {
+                if (yeelightTcpClient.Connected)
+                {
+                    return SendString("{\"id\":" + 1 + ",\"method\":\"set_power\",\"params\":[ " + (status ? "on" : "off") + ", \"sudden\", 100" + "]}\r\n");
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception exc) { lastError = exc; }
+            return false;
+        }
+
+        public bool SetColor(int red, int green, int blue)
+        {
+            try
+            {
+                if (yeelightTcpClient.Connected)
+                {
+                    int value = ((red) << 16) | ((green) << 8) | (blue);
+                    return SendString("{\"id\":" + 1 + ",\"method\":\"set_rgb\",\"params\":[ " + value.ToString() + ", \"sudden\", 100" + "]}\r\n");
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception exc) { lastError = exc; }
+            return false;
+        }
+
+        public bool SetMusicMode(IPAddress localIP, int localPort)
+        {
+            try
+            {
+                if (!musicMode)
+                {
+                    TcpListener musicModeTcpListener;
+                    musicModeTcpListener = new TcpListener(localIP, localPort);
+                    musicModeTcpListener.Start();
+                    SendString("{\"id\":" + 1 + ",\"method\":\"set_music\",\"params\":[1, \"" + localIP + "\"," + localPort + "]}\r\n");
```

Comment:
This entire class looks really good, the only thing I'd change would be to have a function which builds these json strings given id, method, params, etc
Something like
```c#
SendMethod(uint id, string method, params object[] _param) {
    SendString($"\{\"id\":{id}\"method\":{method},\"params\"[{String.Join(", ", _param)}]\}")
}
```